src/main/scala/sigmastate/eval/Evaluation.scala

```diff
@@ -127,15 +132,75 @@ trait Evaluation extends RuntimeCosting { IR: IRContext =>
     case _ => false
   }
 
-  /** Checks if the function (Lambda node) given by the simbol `costF` contains only allowed operations
+  private val _allowedOpCodesInCosting = HashSet[OpCode](
+    Undefined,
+    InputsCode,
+    OutputsCode,
+    TupleCode,
+    SelectFieldCode,
+  )
+
+  def isAllowedOpCodeInCosting(opCode: OpCode): Boolean = {
+    // todo implement
+//    true
+    _allowedOpCodesInCosting.contains(opCode)
+  }
+
+
+  private def getOpCode(d: Def[_]): OpCode = d match {
+      // todo implement
+//    case _: Const[_] => true
+    case _: OpCost | _: PerKbCostOf | _: Cast[_] => Undefined
```

Comment:
Undefined in not a good name for this.

Reply:
@aslesarenko I agree the name `Undefined` is a bit confusing. But it's comment says "  // serialization is not required" which fit this case nicely. Maybe `NotForSerialization` or something similar. How does it sound to you?

Reply:
ok, let's leave Undefined